src/test/java/uk/gov/digital/ho/proving/income/HmrcStub.java

```diff
@@ -2,120 +2,166 @@
 
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
+import uk.gov.digital.ho.proving.income.domain.Applicant;
 
 import java.io.IOException;
 
 import static com.github.tomakehurst.wiremock.client.WireMock.*;
 import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
-import static uk.gov.digital.ho.proving.income.HmrcResponse.MATCH_ID;
 import static uk.gov.digital.ho.proving.income.IpsSearchTest.wireMockRule;
 
 class HmrcStub {
 
     private HmrcResponse hmrcResponse = new HmrcResponse();
 
-    void stubPassUser() throws IOException {
+    void stubPassUser(Applicant applicant) throws IOException {
 
-        getMatchedUser();
+        String applicantId = applicant.nino();
+        getMatchedUser(applicant);
+        wireMockRule.stubFor(get(urlMatching("/individuals/employments/paye\\?matchId=" + applicantId + "&fromDate=[0-9\\-]*&toDate=[0-9\\-]*"))
```

Comment:
There is a lot of duplication in this class... and the stubs are quite hard to read.  Could we move each stub into a private method?  So for example, this method would look something like this:
```
void stubPassUser(Applicant applicant) throws IOException {
    String applicantId = applicant.nino();
    getMatchedUser(applicant);
    stubPaye(applicantId, "/template/payments/pass/employmentsPayeResponse.json");
    stubIncomePaye(applicantId, "/template/payments/pass/incomePayeResponse.json");
    stubSa(applicantId, "/template/payments/pass/incomeSAResponseEmpty.json");
    stubSaSelfEmployment(applicantId, "/template/payments/pass/incomeSASelfEmploymentsResponse.json");
}
```

Reply:
👍